src/main/java/io/appform/dropwizard/discovery/bundle/resolvers/NodeInfoResolver.java

```diff
@@ -0,0 +1,20 @@
+package io.appform.dropwizard.discovery.bundle.resolvers;
+
+import io.appform.dropwizard.discovery.bundle.ServiceDiscoveryConfiguration;
+import io.appform.ranger.common.server.ShardInfo;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+public class NodeInfoResolver implements CriteriaResolver<ShardInfo, ServiceDiscoveryConfiguration>{
```

Comment:
Rename to `DefaultNodeInfoResolver`

Reply:
Again, didn't because, CriteriaResolver is an interface and doesn't necessarily understand NodeInfo, that's why I thought NodeInfoResolver made sense as a name. Or we let go of the interface CriteriaResolver and just write a NodeInfoResolver as an interface with getNode and a DefaultNodeInfoResolver

Reply:
Added a comment on an MR, removing the generic altogether.